pkg/doc/credentialmanifest/credentialmanifest.go

```diff
@@ -0,0 +1,422 @@
+	Title       string
+	Subtitle    string
+	Description string
+	Properties  []interface{}
+}
+
+// UnmarshalJSON is the custom unmarshal function gets called automatically when the standard json.Unmarshal is called.
+func (cm *CredentialManifest) UnmarshalJSON(data []byte) error {
+	err := cm.standardUnmarshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = cm.validate()
+	if err != nil {
+		return fmt.Errorf("invalid credential manifest: %w", err)
+	}
+
+	return nil
+}
+
+// ResolveOutputDescriptors resolves the actual display values for all output descriptors in this credential manifest
+// based on the given vc.
+// vc must be a valid Verifiable Credential in JSON format.
+// It returns an array of ResolvedDataDisplayDescriptors, one for each OutputDescriptor (and in the same order).
+// For each display mapping object, the resolution process is as follows:
+// If the display mapping object uses 1 or more paths, then we try to resolve them one-by-one by looking at vc. We
+// return the first one that resolves successfully. If none of the paths are resolvable, then we return the fallback.
+// If no fallback is specified, then a blank string is returned. This isn't considered an error.
+// If the text field is used instead of paths, then that will simply be returned without needing to look at vc.
+func (cm *CredentialManifest) ResolveOutputDescriptors(vc []byte) ([]ResolvedDataDisplayDescriptor, error) {
+	// The jsonpath library needs the JSON unmarshalled into a map[string]interface{}.
+	vcUnmarshalledIntoMap := map[string]interface{}{}
+
+	err := json.Unmarshal(vc, &vcUnmarshalledIntoMap)
+	if err != nil {
+		return nil, err
+	}
+
+	resolvedDataDisplayDescriptors := make([]ResolvedDataDisplayDescriptor, len(cm.OutputDescriptors))
+
+	for i := range cm.OutputDescriptors {
+		var err error
+
+		resolvedDataDisplayDescriptors[i], err =
+			resolveOutputDescriptor(&cm.OutputDescriptors[i], vcUnmarshalledIntoMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve output descriptors at index %d: %w", i, err)
+		}
+	}
+
+	return resolvedDataDisplayDescriptors, nil
+}
+
+func resolveOutputDescriptor(outputDescriptor *OutputDescriptor,
+	vc map[string]interface{}) (ResolvedDataDisplayDescriptor, error) {
+	var resolvedDataDisplayDescriptor ResolvedDataDisplayDescriptor
+
+	var err error
+
+	resolvedDataDisplayDescriptor.Title,
+		resolvedDataDisplayDescriptor.Subtitle,
+		resolvedDataDisplayDescriptor.Description,
+		err = resolveStaticDisplayMappingObjects(outputDescriptor, vc)
+	if err != nil {
+		return ResolvedDataDisplayDescriptor{}, err
+	}
+
+	resolvedDataDisplayDescriptor.Properties, err =
+		resolveDynamicDisplayMappingObjects(outputDescriptor.Display.Properties, vc)
+	if err != nil {
+		return ResolvedDataDisplayDescriptor{}, err
+	}
+
+	return resolvedDataDisplayDescriptor, nil
+}
+
+func resolveStaticDisplayMappingObjects(outputDescriptor *OutputDescriptor,
+	vc map[string]interface{}) (string, string, string, error) {
```

Comment:
Can a return here be the struct rather than string, string , string ??  like ResolvedDataDisplayDescriptor